File: web/src/dtos/presetMemberPosition.ts
import type { PositionDTO } from "./position";

export interface PresetMemberPositionDTO {
  presetMemberPositionId: number;
  presetMemberId: number;
  positionId: number;
}

export interface PresetMemberPositionDetailDTO extends PresetMemberPositionDTO {
  position: PositionDTO;
}

export interface AddPresetMemberPositionDTO {
  positionId: number;
}

File: python/shared/dtos/preset_member_position.py
from . import BaseDTO, BaseEntityDTO
from .position import PositionDTO


class PresetMemberPositionDTO(BaseEntityDTO):
    preset_member_position_id: int
    preset_member_id: int
    position_id: int


class PresetMemberPositionDetailDTO(PresetMemberPositionDTO):
    position: PositionDTO


class AddPresetMemberPositionDTO(BaseDTO):
    position_id: int
